BlackNectar/BlackNectar/SearchFilterViewController.swift

```diff
@@ -0,0 +1,343 @@
+        
+        annotationView.canShowCallout = true
+        annotationView.leftCalloutAccessoryView = button
+        annotationView.image = blackNectarPin
+        
+        return annotationView
+        
+    }
+    
+}
+
+//MARK: Gets Driving Directions Code
+
+extension SearchFilterViewController {
+    
+    
+    func getDrivingDirections(to storeCoordinates: CLLocationCoordinate2D, with storeName: String) -> MKMapItem {
+        
+        let storePlacemark = MKPlacemark(coordinate: storeCoordinates, addressDictionary: [ "\(title)" : storeName ])
+        let storePin = MKMapItem(placemark: storePlacemark)
+        storePin.name = storeName
+        
+        return storePin
+        
+    }
+    
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView, calloutAccessoryControlTapped control: UIControl) {
+        
+        let appleMapslaunchOptions = [MKLaunchOptionsDirectionsModeKey: MKLaunchOptionsDirectionsModeDriving]
+        
+        if let storeLocation = view.annotation {
+            
+            let storeName: String = (storeLocation.title ?? nil) ?? "Uknown"
+            
+            getDrivingDirections(to: storeLocation.coordinate, with: storeName).openInMaps(launchOptions: appleMapslaunchOptions)
+            
+            AromaClient.beginMessage(withTitle: "User tapped on \(storeName) map pin")
+                .addBody("User navigated to \(storeName)\nstore coordinates: \(storeLocation.coordinate)\n(Search Filter Map View)")
+                .withPriority(.medium)
+                .send()
+            
+        }
+        
+    }
+    
+}
+
+//MARK: Loads Stores When User Pans 
+
+extension SearchFilterViewController {
+    
+    func mapView(_ mapView: MKMapView, regionDidChangeAnimated animated: Bool) {
+        
+        let center = mapView.centerCoordinate
+        
+        LOG.debug("User dragged Map Screen to: \(center)")
+        
+        self.loadStoresInMapView(at: center)
+        
+    }
+    
+}
+
+fileprivate extension MKMapView {
+    
+    func isVisible(annotation: MKAnnotation) -> Bool {
+        
+        let annotationPoint = MKMapPointForCoordinate(annotation.coordinate)
+        
+        return MKMapRectContainsPoint(self.visibleMapRect, annotationPoint)
+    
+    }
+    
+    func removeNonVisibleAnnotations() {
+        
+        self.annotations
+            .filter({ !isVisible(annotation: $0)})
+            .forEach({ self.removeAnnotation($0) })
+        
+    }
+    
+}
+
+//MARK: Aroma Messages
+extension SearchFilterViewController {
+    
+    func makeNoteThatNoStoresFound(additionalMessage: String = "") {
+        
+        LOG.warn("There are no stores around the users location (Stores loading result is 0)")
+        AromaClient.beginMessage(withTitle: "No stores loading result is 0")
+            .addBody("There are no stores around the users location (Stores loading result is 0 : \(additionalMessage)")
```

Comment:
Would be nice to have the location that was searched where no results were found.